sources/packages/backend/apps/api/src/services/application-exception/application-exception.service.ts

```diff
@@ -170,6 +173,15 @@ export class ApplicationExceptionService extends RecordDataModelService<Applicat
           STUDENT_APPLICATION_EXCEPTION_INVALID_STATE,
         );
       }
+      if (
+        exceptionToUpdate.application.applicationStatus ==
+        ApplicationStatus.Overwritten
+      ) {
+        throw new CustomNamedError(
+          `Student application must not be in ${ApplicationStatus.Overwritten} state to be assessed.`,
+          STUDENT_APPLICATION_INVALID_STATE,
+        );
+      }
```

Comment:
Just want to double-check, do we need to check the status separately and throw an error for this scenario?, because, this is an expected scenario, I was thinking we would be restricting the status in the where condition of the query and which will be part of `STUDENT_APPLICATION_EXCEPTION_NOT_FOUND` error

Reply:
On the same page @ann-aot . I believe from the UI overwritten application cannot come for approval. For security sake adding in query would be better unless there is any value.

Reply:
That is for the AC: "Check application status during exception update." 
If the Ministry staff open the UI to approve the exception and meanwhile the user edits the application, the application will have `Overwritten status`.

Reply:
@andrepestana-aot my comment is to add the status restriction in `1` so that it will be part of `3`
![image](https://github.com/bcgov/SIMS/assets/77353155/59a2a393-0925-4e6e-8ece-f6fa342b190d)

Reply:
Do you guys mean removing both `CustomNamedError` (for `ApplicationExceptionStatus.Pending` and `ApplicationStatus.Overwritten`) and adding the condition to the query to ignore them?

Reply:
@andrepestana-aot  my comment is to add the status restriction in `1` so that it will be part of `3`
![image](https://github.com/bcgov/SIMS/assets/77353155/62e13cb9-0b68-4aa3-ae13-9c8071e3e593)